src/network/networkd-state-file.c

```diff
@@ -522,6 +578,60 @@ static void serialize_addresses(
                 fputc('\n', f);
 }
 
+static void serialize_resolvers(
+                FILE *f,
+                const char *lvalue,
+                bool *space,
+                sd_dhcp_lease *lease,
+                bool conditional,
+                sd_dhcp6_lease *lease6,
+                bool conditional6) {
+
+        bool _space = false;
+        if (!space)
+                space = &_space;
+
+        if (lvalue)
+                fprintf(f, "%s=", lvalue);
+
+        if (lease && conditional) {
+                sd_dns_resolver *resolvers;
+                _cleanup_strv_free_ char **names = NULL;
+                int r;
+
+                r = sd_dhcp_lease_get_dnr(lease, &resolvers);
+                if (r < 0)
+                        return (void) log_warning_errno(r, "Failed to get DNR from DHCP lease, ignoring.");
```

Comment:
I can see a lot of messages like
```sh
Oct 22 10:54:38 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
Oct 22 10:54:38 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
Oct 22 10:57:03 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
Oct 22 10:57:57 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
Oct 22 10:57:58 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
Oct 22 10:58:08 default systemd-networkd[297]: Failed to get DNR from DHCP lease, ignoring.
```
when normal DHCPv4 messages with no Encrypted DNS Options are received and it doesn't look right. (I didn't fuzz it or anything like that yet)

Reply:
The message was recently added per https://github.com/systemd/systemd/pull/30952#discussion_r1743325106.